scipy/sparse/_base.py

```diff
@@ -1233,13 +1264,30 @@ def mean(self, axis=None, dtype=None, out=None):
         inter_dtype = np.float64 if integral else res_dtype
         inter_self = self.astype(inter_dtype)
 
+        if type(axis) is tuple and len(axis) == 1:
+            axis = axis[0]
+        
+        # if 1-d, axis should be 0, -1, or None
         if self.ndim == 1:
             if axis not in (None, -1, 0):
                 raise ValueError("axis must be None, -1 or 0")
             res = inter_self / self.shape[-1]
             return res.sum(dtype=res_dtype, out=out)
 
+        # if 2-d and axis is tuple
+        if self.ndim == 2 and type(axis) is tuple:
+            # adjust axes
+            axis = [ax if ax>=0 else ax+2 for ax in axis]
+            if len(axis) == 2:
```

Comment:
Can this be changed similar to above to remove dependence on ndim==2?

Reply:
Done at 8d67717309f9a85f42a7cc182f883c2392f31ef3.